src/webgpu/api/validation/compute_pipeline.spec.ts

```diff
@@ -368,6 +369,97 @@ Tests calling createComputePipeline(Async) validation for uninitialized overrida
     t.doCreateComputePipelineTest(isAsync, _success, descriptor);
   });
 
+g.test('overrides,value,type_error')
+  .desc(
+    `
+Tests calling createComputePipeline(Async) validation for constant values like inf, NaN will results in TypeError.
+`
+  )
+  .params(u =>
+    u //
+      .combine('isAsync', [true, false])
+      .combineWithParams([
+        { constants: { cf: 1 }, _success: true }, // control
+        { constants: { cf: NaN }, _success: false },
+        { constants: { cf: Number.POSITIVE_INFINITY }, _success: false },
+        { constants: { cf: Number.NEGATIVE_INFINITY }, _success: false },
+      ] as { constants: Record<string, GPUPipelineConstantValue>; _success: boolean }[])
+  )
+  .fn(async t => {
+    const { isAsync, constants, _success } = t.params;
+
+    const descriptor = {
+      layout: 'auto' as const,
+      compute: {
+        module: t.device.createShaderModule({
+          code: `
+            override cf: f32 = 0.0;
+            @compute @workgroup_size(1) fn main () {
+              _ = cf;
+            }`,
+        }),
+        entryPoint: 'main',
+        constants,
+      },
+    };
+
+    t.doCreateComputePipelineTest(isAsync, _success, descriptor, 'TypeError');
+  });
+
+g.test('overrides,value,validation_error')
+  .desc(
+    `
+Tests calling createComputePipeline(Async) validation for unrepresentable constant values in compute stage.
+`
+  )
+  .params(u =>
+    u //
+      .combine('isAsync', [true, false])
+      .combineWithParams([
+        { constants: { cu: kValue.u32.min }, _success: true },
+        { constants: { cu: kValue.u32.min - 1 }, _success: false },
+        { constants: { cu: kValue.u32.max }, _success: true },
+        { constants: { cu: kValue.u32.max + 1 }, _success: false },
+        { constants: { ci: kValue.i32.negative.min }, _success: true },
+        { constants: { ci: kValue.i32.negative.min - 1 }, _success: false },
+        { constants: { ci: kValue.i32.positive.max }, _success: true },
+        { constants: { ci: kValue.i32.positive.max + 1 }, _success: false },
+        { constants: { cf: kValue.f32.negative.min }, _success: true },
+        { constants: { cf: -Number.MAX_VALUE }, _success: false },
```

Comment:
We can tighten all of the MAX_VALUE-to-f32 cases. `2**128` should be unambiguously invalid, though we can tighten it even more with `2**128 - 2**103`. That's the midpoint between FLT_MAX and the [special value](https://webidl.spec.whatwg.org/#es-float) of `2**128` which WebIDL says is supposed to be the first invalid value. And I think it should work in our implementation.

Aside: You can write this simply as `2**128 - 2**103` in JS. I verified with BigInt math:
```js
(2n**128n - 2n**104n + 2n**128n) / 2n === BigInt(2**128 - 2**103)
```

Of course we would like to test the next smaller float value (which is supposed to be valid), which is `2**128 - 2**105`, but this will fail in Chrome since we're not sure it's safe with ubsan. Given it's not very important, I say we leave it out for now and add a TODO to test it.

Reply:
Sorry that was wrong. We want to test the next smaller `Number` (`double`) which is `2**128 - 2**104 - 2**75` not `2**128 - 2**105`. Editing

Reply:
```js
const kLastF64CastableToF32     = 340282356779733623858607532500980858880;
const kFirstF64NotCastableToF32 = 340282356779733661637539395458142568448;
```

Reply:
Oh I see. It can be get from removing one fraction bit
`340282356779733661637539395458142568448n - 2n**(0x47en - 1023n) / 2n**52n === 340282356779733623858607532500980858880n`

Reply:
Yeah. I did this the bit-manipulation way:
```
> new BigUint64Array(new Float64Array([2**128 - 2**103]).buffer)[0]
5183643170835005440n
> BigInt(new Float64Array(new BigInt64Array([5183643170835005440n]).buffer)[0])
340282356779733661637539395458142568448n
```